### PR TITLE
Add motor info to QNexuswidget

### PR DESCRIPTION
**Pull request by mspito**
Needs https://github.com/vasole/pymca/pull/1087 first

Screenshot from 2024-08-14 16-15-52

### CODE REVIEW

**Comment by vasole** [comment]
Please rebase your work on current master.

**Comment by vasole** [comment]
Some comments:

PyMca is used at other facilities than the ESRF and former ESRF files did not look like current ones.
I find  `nxpositioners = nxentry.get("instrument/positioners_start", None)` very ESRF-centric

I would aim at using the function getPositionersGroup(h5file, path) in PyMca5.PyMcaCore.NexusTools.py that would make the use of the new feature not restricted to ESRF. That would imply that if one positioner is an array of positions, the first position is used as starting point. A getStartingPositionerValues in PyMca5.PyMcaCore.NexusTools.py seems to me more appropriate than get_motor_positions.

**Comment by mspito** [comment]
I tried to address your suggestions. Is there anything else to be done?

File: src/PyMca5/PyMcaCore/NexusTools.py [hunk 1]
```diff
@@ -510,6 +510,28 @@ def getPositionersGroup(h5file, path):
...
     return positioners
 
+def getStartingPositionersGroup(h5file, path):
+    """
+    Retrieve the start positioners group associated to a path
+    retrieving them from the same entry.
+
+    It assumes they are either in:
+
+    - NXentry/NXinstrument/positioners_start or
```
Comment (woutdenolf) [inline src/PyMca5/PyMcaCore/NexusTools.py:520]:
```suggestion
    This method assumes the positioner group is NXentry/NXinstrument/positioners_start.
```

**Review by woutdenolf — CHANGES_REQUESTED** [review]

**Comment by vasole** [comment]
Prior to it there was a group inside NXinstrument named `positioners` and no `positioners_start`

**Comment by vasole** [comment]
> What makes you think `positioners_start` is "new"? As far as I can remember it has always been there.

If it serves as reminder, `silx convert` is still generating the `NXinstrument/positioners` group.

**Comment by woutdenolf** [comment]
For reference: `NXinstrument/positioners` provides motor positions in the scope of entire scan. This means static motors are scalars and moving motors are arrays. `NXinstrument/positioners_start` and `NXinstrument/positioners_end` are snapshots of motor positions at the start/end of the scan and are therefore scalars.

**Comment by vasole** [comment]
> For reference: `NXinstrument/positioners` provides motor positions in the scope of entire scan. This means static motors are scalars and moving motors are arrays. `NXinstrument/positioners_start` and `NXinstrument/positioners_end` are snapshots of motor positions at the start/end of the scan and are therefore scalars.

`positioners` is always there while start and end are not always there . Just convert a SPEC file with silx convert to verify it. Therefore, I want to check first for positioners_start, and if not present, to use positioners, and if not present to try the Sardana equivalents. It is a 10 minutes work. Simply I do not consider to be my job to do it.

**Comment by woutdenolf** [comment]
So what needs to be done is

- remove `getStartingPositionersGroup` and use `getPositionersGroup` in `get_motor_positions` instead, assuming the datasets can be 0D or 1D.
- modify `getPositionersGroup` to look for `positioners_start` before looking for `positioners`.

Is that correct?

**Comment by vasole** [comment]
One cannot escape to having two different functions (or a function with keywords).

You need to keep into account that to match the SPEC behaviour, you want the motor values associated to the positions when the command was issued. If the information is not available, then one uses the group provided by getPositionersGroup

What you propose should work for this issue, but when I look for the positioners group in PyMca, I actually want the group containing the scanned ones because I want the positions during the scan and not when the command was issued.

So, I would just extend the getStartingPositionersGroup to handle the case positioners_start is not there and it gets the getPositionersGroup instead. The handling of 1D data instead of 0D would indeed be necessary. If you want to centralize the dimensions handling as getStartingPositionersAndValues, I am pretty sure it can avoid duplicating code somewhere else.

**Comment by vasole** [comment]
> We added the function _getStartingPositionerValues_ which calls the function _getStartingPositionersGroup_ which falls back to the function _getPositionersGroup_. Is that what you had in mind to be more generic ?

Yes, it is. Please, just make the new functionality available just at NXentry level.

**Comment by woutdenolf** [comment]
We thought it would be easier if you could click anywhere in the scan and see the list of motors.

**Comment by mspito** [comment]
Now we can access to the motor positions only at NXentry level. Is that what you expected ?

**Pull request merged**